app/models/path_redirect.rb

```diff
@@ -0,0 +1,25 @@
+class PathRedirect < ApplicationRecord
+  SOURCES = %w[admin service].freeze
+
+  validates :old_path, presence: true, uniqueness: true
+  validates :new_path, presence: true
+  validates :source, presence: true, inclusion: { in: SOURCES }
+
+  before_save :increment_version, if: :will_save_change_to_new_path?
+
+  resourcify
+
+  def old_path_url
+    URL.url(old_path)
+  end
+
+  def new_path_url
+    URL.url(new_path)
+  end
+
+  private
+
+  def increment_version
+    self.version += 1
```

Comment:
The `self` is redundant here, no?
```suggestion
    version += 1
```

Reply:
No, it has to be `self` when setting the value to refer to the attribute on the record itself. Without `self` Ruby will interpret as trying to set/modify a local variable of `version` which isn't defined and results in:
```ruby
NoMethodError: undefined method `+' for nil:NilClas
```